api/v1beta1/azuremachine_validation_test.go

```diff
@@ -868,3 +868,51 @@ func TestAzureMachine_ValidateNetwork(t *testing.T) {
 		})
 	}
 }
+
+func TestAzureMachine_ValidateDiagnostics(t *testing.T) {
```

Comment:
Can you add a test case to ensure there are no expected errors if diagnostics or diagnostics.Boot is nil?

Reply:
shall i just return an error statement if either one of them is nil?

Reply:
You don't need to return an error statement. Since we don't expect an error, you can just set `expectedErrors: nil` like you did here https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3498/files#diff-c83d675c43071dba2ff5635eba4200fc6782df2b85c518ab5872f8679a4185b0R884